File: app/src/main/java/com/example/david/lists/view/addedit/userlist/buildlogic/AddEditUserListDialogModule.java
package com.example.david.lists.view.addedit.userlist.buildlogic;

import com.example.david.lists.common.buildlogic.ViewScope;
import com.example.david.lists.data.repository.IRepositoryContract;
import com.example.david.lists.util.ISchedulerProviderContract;
import com.example.david.lists.view.addedit.common.IAddEditContract;
import com.example.david.lists.view.addedit.userlist.AddEditUserListLogic;

import javax.inject.Named;

import dagger.Module;
import dagger.Provides;
import io.reactivex.disposables.CompositeDisposable;

import static com.example.david.lists.view.addedit.common.buildlogic.AddEditCommonNamedConstants.ID;
import static com.example.david.lists.view.addedit.common.buildlogic.AddEditCommonNamedConstants.TITLE;

@Module
final class AddEditUserListDialogModule {
    @ViewScope
    @Provides
    IAddEditContract.Logic logic(IAddEditContract.View view,
                                 IAddEditContract.ViewModel viewModel,
                                 IRepositoryContract.Repository repository,
                                 ISchedulerProviderContract schedulerProvider,
                                 CompositeDisposable disposable,
                                 @Named(ID) String id,
                                 @Named(TITLE) String title) {
        return new AddEditUserListLogic(view, viewModel, repository, schedulerProvider, disposable, id, title);
    }
}

File: app/src/main/java/com/example/david/lists/view/addedit/userlist/buildlogic/AddEditUserListDialogModule.kt
package com.example.david.lists.view.addedit.userlist.buildlogic

import com.example.david.lists.common.buildlogic.ViewScope
import com.example.david.lists.util.ISchedulerProviderContract
import com.example.david.lists.view.addedit.common.IAddEditContract
import com.example.david.lists.view.addedit.common.buildlogic.AddEditCommonNamedConstants.ID
import com.example.david.lists.view.addedit.common.buildlogic.AddEditCommonNamedConstants.POSITION
import com.example.david.lists.view.addedit.common.buildlogic.AddEditCommonNamedConstants.TITLE
import com.example.david.lists.view.addedit.userlist.AddEditUserListLogic
import com.example.domain.repository.IRepositoryContract
import dagger.Module
import dagger.Provides
import javax.inject.Named

@Module
internal class AddEditUserListDialogModule {
    @ViewScope
    @Provides
    fun logic(view: IAddEditContract.View,
              viewModel: IAddEditContract.ViewModel,
              repository: IRepositoryContract.Repository,
              schedulerProvider: ISchedulerProviderContract,
              @Named(ID) id: String,
              @Named(TITLE) title: String,
              @Named(POSITION) position: Int): IAddEditContract.Logic {
        return AddEditUserListLogic(view, viewModel, repository, schedulerProvider, id, title, position)
    }
}
